src/traits/networking/centralized_web_server_network.rs

```diff
@@ -0,0 +1,720 @@
+                            // Wait for the view to change before polling for proposals again
+                            consensus_info = consensus_update.recv().await.unwrap();
+                        }
+                        MessageType::VoteTimedOut => {
+                            info!(
+                                "Received {} votes for view {}",
+                                deserialized_messages.len(),
+                                consensus_info.view_number
+                            );
+                            let mut direct_poll_queue = self.direct_poll_queue.write().await;
+                            for vote in &deserialized_messages {
+                                vote_index += 1;
+                                direct_poll_queue.push(vote.clone());
+                            }
+                        }
+                        MessageType::Transaction => {
+                            info!("Received new transaction");
+                            let mut lock = self.broadcast_poll_queue.write().await;
+                            for tx in &deserialized_messages {
+                                tx_index += 1;
+                                lock.push(tx.clone());
+                            }
+                        }
+                    }
+                }
+                Ok(None) => {
+                    async_sleep(self.wait_between_polls).await;
+                }
+                Err(_e) => {
+                    async_sleep(self.wait_between_polls).await;
+                }
+            }
+            // Check if there is updated consensus info
+            let new_consensus_info = consensus_update.try_recv();
+            if new_consensus_info.is_ok() {
+                consensus_info = new_consensus_info.unwrap();
+                vote_index = 0;
+            }
+        }
+    }
+
+    /// Sends a GET request to the webserver for some specified endpoint
+    /// Returns a vec of deserialized, received messages or an error
+    async fn get_message_from_web_server(
+        &self,
+        endpoint: String,
+    ) -> Result<Option<Vec<RecvMsg<M>>>, ClientError> {
+        let result: Result<Option<Vec<Vec<u8>>>, ClientError> =
+            self.client.get(&endpoint).send().await;
+        match result {
+            Err(error) => Err(error),
+            Ok(Some(messages)) => {
+                let mut deserialized_messages = Vec::new();
+                for message in &messages {
+                    let deserialized_message = bincode::deserialize(message).unwrap();
+                    deserialized_messages.push(deserialized_message);
+                }
+                Ok(Some(deserialized_messages))
+            }
+            Ok(None) => Ok(None),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(bound(deserialize = ""))]
+/// A message being sent to the web server
+pub struct SendMsg<M: NetworkMsg> {
```

Comment:
Nice! Great job working around the (admittedly brittle) networking APIs with `SendMsg` and `RecvMsg`!